main/main.ino

```diff
@@ -1613,6 +1622,65 @@ void receivingMQTT(char* topicOri, char* datacallback) {
   }
 }
 
+#ifdef MQTT_HTTPS_FW_UPDATE
+#  include <WiFiClientSecure.h>
+#  ifdef ESP32
+#    include <HTTPUpdate.h>
+#  elif ESP8266
+#    include <ESP8266httpUpdate.h>
+#  endif
+void MQTTHttpsFWUpdate(char* topicOri, JsonObject& HttpsFwUpdateData) {
+  if (strstr(topicOri, subjectFWUpdate) != NULL) {
+    if (strcmp(HttpsFwUpdateData["version"], OMG_VERSION) != 0) {
+      const char* url = HttpsFwUpdateData["url"];
+      if (!strstr((url + strlen(url - 5)), ".bin")) {
+        Log.error(F("Invalid firmware extension" CR));
+        return;
+      }
+
+      Log.warning(F("Starting firmware update" CR));
+      WiFiClientSecure update_client;
+#  ifdef ESP32
+#    ifndef NTP_SERVER
+      update_client.setInsecure();
+#    else
+      if (strlen(https_fw_server_cert) < 200) {
+        Log.error(F("Invalid certificate - using insecure" CR));
+        update_client.setInsecure();
+      } else {
+        update_client.setCACert(https_fw_server_cert);
+      }
+#    endif
+      update_client.setTimeout(12);
+      /****** For Future use, does not work with current releases of arduino core ******/
+      // httpUpdate.setFollowRedirects(HTTPC_STRICT_FOLLOW_REDIRECTS);
```

Comment:
To work around this issue for older core (fix is only a month old) I could derive a class to incorporate the fix. Would need extra files or a separate library.